src/Core/Grid/View/GridViewFactory.php

```diff
@@ -0,0 +1,252 @@
+                continue;
+            }
+
+            $actions[$rowAction->getIdentifier()] = [
+                'name' => $rowAction->getName(),
+                'icon' => $rowAction->getIcon(),
+                'url' => $url,
+            ];
+        }
+
+        return $actions;
+    }
+
+    /**
+     * Some columns may modify row data that comes directly from database or any other data source.
+     *
+     * @param array                     $row
+     * @param ColumnCollectionInterface $columns
+     *
+     * @return array
+     */
+    private function applyColumnModifications(array $row, ColumnCollectionInterface $columns)
+    {
+        foreach ($columns as $column) {
+            // if for some reason column does not exist in a row
+            // then let developer know that something is wrong
+            if (!isset($row[$column->getIdentifier()])) {
+                throw new MissingColumnInRowException(
+                    sprintf('Column "%s" does not exist in row "%s"', $column->getIdentifier(), json_encode($row))
+                );
+            }
+
+            // if column does not modify data
+            // then we keep original column data and skip modification
+            if (!is_callable($column->getModifier())) {
+                continue;
+            }
+
+            $row[$column->getIdentifier()] = call_user_func($column->getModifier(), $row);
+        }
+
+        return $row;
+    }
+
+    /**
+     * @param ColumnCollectionInterface $columns
+     *
+     * @return array
+     */
+    private function createColumnsView(ColumnCollectionInterface $columns)
+    {
+        $columnsView = [];
+
+        foreach ($columns as $column) {
+            $columnsView[] = [
+                'identifier' => $column->getIdentifier(),
+                'name' => $column->getName(),
+                'is_sortable' => $column->isSortable(),
+                'is_filterable' => $column->isFilterable(),
+                'is_raw' => $column->isRawContent(),
+            ];
+        }
+
+        return $columnsView;
+    }
+
+    /**
+     * Builds filters form for grid
+     *
+     * @param GridDefinitionInterface $definition
+     *
+     * @return FormView
+     */
+    private function createFilterFormView(GridDefinitionInterface $definition)
+    {
+        $formBuilder = $this->formFactory->createNamedBuilder($definition->getIdentifier());
+
+        foreach ($definition->getColumns() as $column) {
+            if ($formType = $column->getFilterFormType()) {
+                $options = $column->getFilterFormTypeOptions();
+
+                if (!isset($options['required'])) {
+                    $options['required'] = false;
+                }
+
+                $formBuilder->add(
+                    $column->getIdentifier(),
+                    $formType,
+                    $options
+                );
+            }
+        }
+
+        $form = $formBuilder
+            ->setData([])   //@todo: filters data should be
```

Comment:
duly noted, after merging this contribution I'll rebase mine and I will complete.